show_client/mac_cli.go

```diff
@@ -46,3 +48,113 @@ func getMacAgingTime(options sdc.OptionMap) ([]byte, error) {
 	}
 	return json.Marshal(result)
 }
+
+// MacEntry represents a single FDB entry
+type MacEntry struct {
+	Vlan       string `json:"vlan"`
+	MacAddress string `json:"macAddress"`
+	Port       string `json:"port"`
+	Type       string `json:"type"`
+}
+
+const (
+	ApplDb   = "APPL_DB"
+	StateDb  = "STATE_DB"
+	FDBTable = "FDB_TABLE"
+)
+
+// getMacTable queries APPL_DB and STATE_DB FDB_TABLE entries and returns either the list or count per options
+func getMacTable(options sdc.OptionMap) ([]byte, error) {
+	// Parse filters
+	vlanFilter := -1
+	if v, ok := options["vlan"].Int(); ok {
+		vlanFilter = v
+	}
+	portFilter, _ := options["port"].String()
+	addrFilter, _ := options["address"].String()
+	typeFilter, _ := options["type"].String()
+	wantCount, _ := options["count"].Bool()
+
+	// Fetch APPL_DB and STATE_DB separately to preserve source and avoid key collisions
+	applData, err := GetMapFromQueries([][]string{{ApplDb, FDBTable}})
+	if err != nil {
+		log.Errorf("Unable to get APPL_DB FDB_TABLE, err: %v", err)
+		return nil, err
+	}
+	stateData, err := GetMapFromQueries([][]string{{StateDb, FDBTable}})
+	if err != nil {
+		log.Errorf("Unable to get STATE_DB FDB_TABLE, err: %v", err)
+		return nil, err
+	}
+
+	// Prefer APPL_DB entries on duplicates; track seen keys "vlan|mac"
+	seen := make(map[string]struct{})
+	entries := make([]MacEntry, 0, len(applData)+len(stateData))
+
+	addIfMatch := func(vlan, macAddress, port, mtype string) {
+		// Filters
+		if vlanFilter >= 0 && vlan != fmt.Sprint(vlanFilter) {
+			return
+		}
+		if portFilter != "" && !strings.EqualFold(port, portFilter) {
+			return
+		}
+		if addrFilter != "" && !strings.EqualFold(strings.ToLower(addrFilter), strings.ToLower(macAddress)) {
+			return
+		}
+		if typeFilter != "" && strings.ToLower(typeFilter) != strings.ToLower(mtype) {
```

Comment:
if type filter is not one of the expected one like "dynamic" we should also return an error.